tools/ci/tc/decision.py

```diff
@@ -0,0 +1,314 @@
+        return jobs
+
+    regexp = re.compile(r"\s*tc-jobs:(.*)$")
+
+    for line in body.splitlines():
+        m = regexp.match(line)
+        if m:
+            items = m.group(1)
+            for item in items.split(","):
+                jobs.add(item.strip())
+            break
+    return jobs
+
+
+def filter_schedule_if(event, tasks):
+    scheduled = {}
+    run_jobs = None
+    for name, task in iteritems(tasks):
+        if "schedule-if" in task:
+            if "run-job" in task["schedule-if"]:
+                if run_jobs is None:
+                    run_jobs = get_run_jobs(event)
+                if "all" in run_jobs or any(item in run_jobs for item in task["schedule-if"]["run-job"]):
+                    scheduled[name] = task
+        else:
+            scheduled[name] = task
+    logger.info("Scheduling rules match tasks:\n * %s" % "\n * ".join(scheduled.keys()))
+    return scheduled
+
+
+def get_fetch_rev(event):
+    is_pr, _ = get_triggers(event)
+    if is_pr:
+        # Try to get the actual rev so that all non-decision tasks are pinned to that
+        ref = "refs/pull/%s/merge" % event["pull_request"]["number"]
+        try:
+            output = subprocess.check_output(["git", "ls-remote", "origin", ref])
+        except subprocess.CalledProcessError:
+            import traceback
+            logger.error(traceback.format_exc())
+            logger.error("Failed to get merge commit sha1")
+            return ref, None
+        if not output:
+            logger.error("Failed to get merge commit")
+            return ref, None
+        return ref, output.split()[0]
+    else:
+        return event["ref"], event["after"]
+
+
+def build_full_command(event, task):
+    fetch_ref, fetch_rev = get_fetch_rev(event)
+    cmd_args = {
+        "task_name": task["name"],
+        "repo_url": event["repository"]["clone_url"],
+        "fetch_ref": fetch_ref,
+        "task_cmd": task["command"],
+        "install_str": "",
+    }
+
+    options = task.get("options", {})
+    options_args = []
+    if fetch_rev is not None:
+        options_args.append("--rev=%s" % fetch_rev)
+    if options.get("oom-killer"):
+        options_args.append("--oom-killer")
+    if options.get("xvfb"):
+        options_args.append("--xvfb")
+    if not options.get("hosts"):
+        options_args.append("--no-hosts")
+    else:
+        options_args.append("--hosts")
+    if options.get("checkout"):
+        options_args.append("--checkout=%s" % options["checkout"])
+    for browser in options.get("browser", []):
+        options_args.append("--browser=%s" % browser)
+    if options.get("channel"):
+        options_args.append("--channel=%s" % options["channel"])
+    if options.get("checkout"):
+        options_args.append("--checkout=%s" % options["checkout"])
+    if options.get("install-certificates"):
+        options_args.append("--install-certificates")
+
+    cmd_args["options_str"] = " ".join("%s" % item for item in options_args)
```

Comment:
```suggestion
    cmd_args["options_str"] = " ".join(str(item) for item in options_args)
```

Or are the semantics different? My suggestion reads better IMHO :)